File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currsum = nums[0];
        int maxsum = nums[0];

        for(int i = 1; i <= nums.length-1; i++) {
            if(currsum + nums[i] > nums[i]) {
                currsum += nums[i];
            }
            else {
                currsum = nums[i];
            }
            maxsum = Math.max(maxsum, currsum);
        }
        return maxsum;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curSum = nums[0]
        maxSum = nums[0]

        for i in range(1, len(nums)):
            curSum = max(nums[i], curSum + nums[i])
            maxSum = max(maxSum, curSum)

        return maxSum



        # maxSum = nums[0]
        # n = len(nums)

        # for i in range(n):
        #     curSum = 0
        #     for j in range(i, n):
        #         curSum += nums[j]
        #         maxSum = max(maxSum, curSum)

        # return maxSum